shiny/render/_render.py

```diff
@@ -96,35 +101,75 @@ def text(
 # ======================================================================================
 # RenderPlot
 # ======================================================================================
+
+
+def _default_plot_ui(
+    id: str, *args: Any, _params: dict[str, Any] | None = None, **kwargs: Any
+) -> Tag:
+    # The _params argument contains the arguments that were passed to `@render.plot`. In
+    # the specific case of `@render.plot`, we grab the width and height arguments from
+    # the decorator and apply them to the output_plot call.
+    if _params is not None:
+        extra_args = {}
+
+        if "width" in _params:
+            extra_args["width"] = (
+                f"{_params['width']}px"
+                if isinstance(_params["width"], (int, float))
+                else str(_params["width"])
+            )
+
+        if "height" in _params:
+            extra_args["height"] = (
+                f"{_params['height']}px"
+                if isinstance(_params["height"], (int, float))
+                else str(_params["height"])
+            )
+
+        kwargs.update(extra_args)
+
+    return _ui.output_plot(id, *args, **kwargs)
+
+
 # It would be nice to specify the return type of RenderPlotFunc to be something like:
 #   Union[matplotlib.figure.Figure, PIL.Image.Image]
 # However, if we did that, we'd have to import those modules at load time, which adds
 # a nontrivial amount of overhead. So for now, we're just using `object`.
-@output_transformer(default_ui=_ui.output_plot)
+@output_transformer(default_ui=_default_plot_ui)
 async def PlotTransformer(
     _meta: TransformerMetadata,
     _fn: ValueFn[object],
     *,
     alt: Optional[str] = None,
+    width: Optional[float] = None,
+    height: Optional[float] = None,
     **kwargs: object,
 ) -> ImgData | None:
     is_userfn_async = is_async_callable(_fn)
     name = _meta.name
     session = _meta.session
 
-    ppi: float = 96
-
     inputs = session.root_scope().input
 
     # Reactively read some information about the plot.
     pixelratio: float = typing.cast(
         float, inputs[ResolvedId(".clientdata_pixelratio")]()
     )
-    width: float = typing.cast(
-        float, inputs[ResolvedId(f".clientdata_output_{name}_width")]()
-    )
-    height: float = typing.cast(
-        float, inputs[ResolvedId(f".clientdata_output_{name}_height")]()
+
+    # Do NOT call this unless you actually are going to respect the container dimension
+    # you're asking for. It takes a reactive dependency. If the client hasn't reported
+    # the requested dimension, you'll get a SilentException.
+    def container_size(dimension: Literal["width", "height"]) -> float:
+        result = inputs[ResolvedId(f".clientdata_output_{name}_{dimension}")]()
+        return typing.cast(float, result)
+
+    plot_size_info = PlotSizeInfo(
```

Comment:
Please add comment explaining that we don't have enough information at this point to be sure of size, so we're going to build up that information.